lib/openshift/resource.rb

```diff
@@ -286,8 +286,8 @@ def conditions(user: nil, cached: true, quiet: false)
 
     # returns a hash simliar to
     # {"lastTransitionTime"=>2019-03-23 07:33:34 UTC, "status"=>"False", "type"=>"Progressing"}
-    def condition(type: nil, user: nil, cached: true, quiet: false)
-      res = self.conditions.select { |c| c['type'] == type }
+    def condition(type: nil, user: nil, cached: false, quiet: false)
```

Comment:
can we plese default to `true`?

Reply:
if we rely on defaults in called methods we end up with missing parameter passing as it seems the case ;)

Reply:
@akostadinov  I prefer to have it `false` because it’s a dynamic value…don’t want to assume the caller of the method to set that value.  For example https://github.com/openshift/verification-tests/blob/master/lib/openshift/resource.rb#L261

Reply:
@akostadinov as an actual example, @anuragthehatter was using the method and was getting staled data currently.

Reply:
everything is dynamyc. Most everything is dynamic. He must remember that values are cached unless enforced.

Reply:
true, some are more dynamic than others. :)

Reply:
See, in the beginning tried to differentiate and at the end it was a mess. pls take as a requirement that values are cached by default. Wait methods should hardcode `cached: false`.

Reply:
usually makes scenario inline checks shorter also

Reply:
updated...PTAL... thx!

Reply:
thank you!

Reply:
Thanks @pruan-rht @akostadinov. Enforcing cached wherever needed sounds good to me. Defaulted to true is okay.